code/graphics/opengl/gropenglshader.cpp

```diff
@@ -564,6 +567,20 @@ static bool load_cached_shader_binary(opengl::ShaderProgram* program, const SCP_
 	auto binary_format = (GLenum) format;
 	json_decref(metadata_root);
 
+	bool supported = false;
+	for (auto supported_fmt : GL_binary_formats) {
+		if (supported_fmt == binary_format) {
```

Comment:
This comparison is apparently between a signed and an unsigned value; `GL_binary_formats` is a vector of `GLint`s while `binary_format` seems to be a `GLenum`; shouldn't those two types match? If nothing else, this comparison needs to not generate a warning for CI to pass.

Reply:
`GL_binary_formats` is initialized from an OpenGL function which only returns `GLint` so that's the reason why it's not a vector of `GLenum`.

I added a cast to silence the warning.